lib/matplotlib/colors.py

```diff
@@ -690,6 +690,16 @@ def __copy__(self):
         cmapobject._global = False
         return cmapobject
 
+    def __eq__(self, other):
+        if not isinstance(other, Colormap):
+            return False
+        # To compare lookup tables the Colormaps have to be initialized
+        if not self._isinit:
+            self._init()
+        if not other._isinit:
+            other._init()
+        return np.array_equal(self._lut, other._lut)
```

Comment:
colormaps know their own name, is that something we want to also check?

Reply:
Yes, IMHO we should. As we use it now, the name is an integral part of the colormap. Otherwise you can get funny behavior, e.g. with the new colormap registry
```
cmap in colormaps.values()  # True
cmap.name in colormaps.keys()  # False
```
And similar.

Reply:
I don't have a strong opinion either way. I was viewing the `==` to mean "will these colormaps look the same". If cmocean defines a "Blues1" colormap and it looks suspiciously similar to MPL's "Blues", I may want to check for that equality and not name equality. I do see TIm's point though, so I am fine adding it if that is what is wanted.

I suppose we should also add `colorbar_extends` too in case someone changed that.

Reply:
I see the point in "will these colormaps look the same", but as long as we have `Colormap.name` I think it has to be taken into account for `__eq__`. You could introduce a `Colorbar.is_equivalent()` for "look the same". OTOH I don't think there is a strong reason a colormap has to know its name, but removing that is hard to justify as well.

Reply:
OK, I added in the comparisons for `name` and `colorbar_extend`. Also added tests for those two situations.